exemplar/src/main/kotlin/com/squareup/exemplar/actions/ClockWebSocketAction.kt

```diff
@@ -0,0 +1,37 @@
+package com.squareup.exemplar.actions
+
+import misk.web.ConnectWebSocket
+import misk.web.actions.WebAction
+import misk.web.actions.WebSocket
+import misk.web.actions.WebSocketListener
+import java.util.Date
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ClockWebSocketAction : WebAction {
+  @Inject private lateinit var executorService: ScheduledExecutorService
+
+  @ConnectWebSocket("/clock")
```

Comment:
Why does ``@ConnectWebSocket`` use annotation based dispatch but the rest of the API use a statically typed listener interface? Feels inconsistent and a little jarring to switch between the two modes of API. I'd propose either making all of websockets support be listener interface based, or use annotations for dispatching the other types of web socket events (Closing, Closed, OnMessage) in addition to the connect.

Reply:
My model is that a websocket is a special interactive response. Instead of accepting a value as the request body you accept a handle to send messages. Instead of returning a value as the response body you return a handle to receive messages.